include/boost/gil/image_processing/morphology.hpp

```diff
@@ -0,0 +1,271 @@
+//
+// Copyright 2021 Prathamesh Tagore <prathameshtagore@gmail.com>
+//
+// Use, modification and distribution are subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+//
+
+#ifndef BOOST_GIL_IMAGE_PROCESSING_MORPHOLOGY_HPP
+#define BOOST_GIL_IMAGE_PROCESSING_MORPHOLOGY_HPP
+#include <boost/gil/extension/numeric/kernel.hpp>
+#include <boost/gil/gray.hpp>
+#include <boost/gil/image_processing/threshold.hpp>
+
+namespace boost {
+namespace gil {
+enum class morphological_operation {
+  dilation,
+  erosion,
+};
```

Comment:
Is this used outside `detail` namespace? If not, please move inside that namespace.